src/vs/platform/terminal/common/terminalProcess.ts

```diff
@@ -79,7 +79,13 @@ export interface ISerializedCommand {
 	exitCode: number | undefined;
 	commandStartLineContent: string | undefined;
 	timestamp: number;
+	genericProperties?: IGenericCommandProperties;
 }
+
+export interface IGenericCommandProperties {
+	hoverMessage: string;
```

Comment:
I'd make this string | undefined and pass in undefined for the regular `ITermOscPt.SetMark` sequence

Reply:
since it turns into a string when sent as a sequence, it becomes "undefined", which i'll have to compare against here https://github.com/microsoft/vscode/blob/be9e6021ea9be9da1aa22f7c8a8c79e734db412f/src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts#L320

Reply:
I think cleaner to let this check fail with `''`, but yeah changed the type to allow undefined